sp-vitl03

lukas-vit opened the pull request:
Odevzdávám pilotní commit semestrální práce
Připojení do databáze, základní layout, sign in.
Dostupné z: https://eso.vse.cz/~vitl03/sp_pilot/

lukas-vit committed:
sp-vitl03

nvbach91 reviewed (commented):
doufam ze nebudete prejmenovavat slozku sp_pilot

lukas-vit committed:
aktualizace sp

lukas-vit committed:
dokoncena sp

lukas-vit committed:
fixed errors

nvbach91 commented on www/vitl03/sp_pilot/checkout.php:
```diff
@@ -0,0 +1,310 @@
+<?php include __DIR__ . '/includes/header.php' ?>
+
+<div class="section">
+	<div class="container">
```
indentace vsech kodu 4 mezery

nvbach91 commented on www/vitl03/sp_pilot/class/ProductsDB.php:
```diff
@@ -0,0 +1,183 @@
+    {
+
+        $statement = $this->pdo->prepare('SELECT * FROM ' . $this->tableName . ' WHERE product_id=?');
+        $statement->execute(([$_GET['id']]));
```
id predavejte pres argument funkce, ne pres globalni GET (ve vice mistech v kodu)

nvbach91 commented on www/vitl03/sp_pilot/class/ProductsDB.php:
```diff
@@ -0,0 +1,183 @@
+    }
+    public function fetchAllPagination($nItemsPerPagination,$offset)
+    {
+        $statement = $this->pdo->prepare('SELECT * FROM ' . $this->tableName . ' WHERE 1 LIMIT '.$nItemsPerPagination.' OFFSET ?;');
```
taky by neuskodilo ten limit davat do execute

nvbach91 commented on www/vitl03/sp_pilot/class/ProductsDB.php:
```diff
@@ -0,0 +1,183 @@
+        $products_in_cart = isset($_SESSION['cart']) ? $_SESSION['cart'] : array();
+        $statement = $this->pdo->prepare('SELECT * FROM ' . $this->tableName . ' WHERE product_id IN (' . $array . ')');
+        $statement->execute(array_keys($products_in_cart));
+        return $statement->fetchAll(PDO::FETCH_ASSOC);
```
FETCH_ASSOC uz by mel byt nastaveno ve tride Database

nvbach91 commented on www/vitl03/sp_pilot/class/UsersDB.php:
```diff
@@ -0,0 +1,71 @@
+    }
+    public function update($value, $key)
+    {
+        $statement = $this->pdo->prepare('UPDATE ' . $this->tableName . ' SET privillage=? WHERE id=?');
```
tusim ze se to pise privilege

nvbach91 commented on www/vitl03/sp_pilot/create-item.php:
```diff
@@ -0,0 +1,50 @@
+<?php include __DIR__ . '/includes/header.php' ?>
+
+
+<div class="container" style="margin-bottom: 200px;">
```
css pouze pres `<link>`

nvbach91 commented on www/vitl03/sp_pilot/database_connection.php:
```diff
@@ -0,0 +1,7 @@
+const DB_HOST = 'localhost';
+const DB_DATABASE = 'vitl03';
+const DB_USERNAME = 'vitl03';
+const DB_PASSWORD = 'eigheeLae4Aith9aiH';
```
neni vhodne ukazovat heslo

nvbach91 commented on www/vitl03/sp_pilot/database_connection.php:
```diff
@@ -0,0 +1,7 @@
+<?php
```
tenhle soubor by se mel jmenovat jinak

nvbach91 commented on www/vitl03/sp_pilot/delete.php:
```diff
@@ -0,0 +1,8 @@
+session_start();
+$productsDB = new ProductsDB();
+$productsDB->delete();
+
```
tady by mela byt osetrovani vstupnich dat z globalniho objektu GET nebo POST, ne ve tride ProductsDB

nvbach91 commented on www/vitl03/sp_pilot/includes/footer.php:
```diff
@@ -0,0 +1,85 @@
+$categories = $categoriesDB->fetchAll2();
+?>
+<footer id="footer">
+	<div class="section">
```
indentace 4 mezery

nvbach91 commented on www/vitl03/sp_pilot/includes/footer.php:
```diff
@@ -0,0 +1,85 @@
+
+
+
+<script src="js/jquery.min.js"></script>
```
import js/css z CDN

nvbach91 commented on www/vitl03/sp_pilot/orders.php:
```diff
@@ -0,0 +1,88 @@
+
+
+$ordersDB = new OrdersDB();
+$orders = $ordersDB->fetchByEmail2($userEmail);
```
fetchAll2, fetchByEmail2, ... prejmenujte na verzi bez cislice, a to co neplati zakomentujte

lukas-vit committed:
opraveny chyby

opraveno: javascript z CDN, v metodách již nejsou čísla, odebrán soubor database_connection, DatabaseOperations, id předáváno přes argument, privallage přejmenováno na privilege, ošetření vstupních dat

lukas-vit committed:
oprava

nvbach91 reviewed (approved):
![image](https://user-images.githubusercontent.com/20724910/119961588-83b3b100-bfa6-11eb-8b97-338c5fb48df5.png)
